components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java

```diff
@@ -88,24 +97,58 @@ public static URI buildURIForBody(String endpoint) {
      */
     public static URI buildURIForHeader(String endpoint) {
 
-        String tenantQualifiedRelativePath =
-                String.format(TENANT_CONTEXT_PATH_COMPONENT, getTenantDomainFromContext()) + SERVER_API_PATH_COMPONENT;
-        String url = IdentityUtil.getEndpointURIPath(tenantQualifiedRelativePath + endpoint, false, true);
-
-        URI loc = URI.create(url);
-        if (!loc.isAbsolute()) {
-            Message currentMessage = PhaseInterceptorChain.getCurrentMessage();
-            if (currentMessage != null) {
-                UriInfo ui = new UriInfoImpl(currentMessage.getExchange().getInMessage(), null);
-                try {
-                    return new URI(ui.getBaseUri().getScheme(), ui.getBaseUri().getAuthority(), url, null, null);
-                } catch (URISyntaxException e) {
-                    LOG.error("Server encountered an error while building the location URL with scheme: " +
-                            ui.getBaseUri().getScheme() + ", authority: " + ui.getBaseUri().getAuthority() +
-                            ", url: " + url, e);
-                }
-            }
+        URI loc;
+        String context = getContext(endpoint);
+
+        try {
+            String url = ServiceURLBuilder.create().addPath(context).build().getAbsolutePublicURL();
+            loc = URI.create(url);
+        } catch (URLBuilderException e) {
+            String errorDescription = "Server encountered an error while building URL for response header.";
+            throw buildInternalServerError(e, errorDescription);
         }
         return loc;
     }
+
+    /**
+     * Builds the API context on whether the tenant qualified url is enabled or not. In tenant qualified mode the
+     * ServiceURLBuilder appends the tenant domain to the URI as a path param automatically. But
+     * in non tenant qualified mode we need to append the tenant domain to the path manually.
+     *
+     * @param endpoint Relative endpoint path.
+     * @return Context of the API.
+     */
+    private static String getContext(String endpoint) {
+
+        String context;
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            context = SERVER_API_PATH_COMPONENT + endpoint;
+        } else {
+            context = String.format(TENANT_CONTEXT_PATH_COMPONENT, getTenantDomainFromContext()) +
+                    SERVER_API_PATH_COMPONENT + endpoint;
+        }
+        return context;
+    }
+
+    /**
+     * Builds APIError to be thrown if the URL building fails.
+     *
+     * @param e Exception occurred while building the URL.
```

Comment:
Shall we fix the formatting errors here?

Reply:
Can you please mention the exact formatting error here?